boards/arm/nrf52810_pca10040/board.h

```diff
@@ -9,27 +9,4 @@
 
 #include <soc.h>
 
-/* Aliases to make the basic samples (e.g. LED, button) work */
-#define LED0_GPIO_PIN GPIO_LEDS_0_GPIO_PIN
```

Comment:
How are these macros defined now?
Basic examples (e.g. blinky) still need  macros such as: LED0_GPIO_PIN

Reply:
@ioannisg I think I can answer. I had the same question and just ran the build. All these defines are now generated in the generated....h file:
/* button@0 */
#define GPIO_KEYS_0_GPIO_CONTROLLER     "GPIO_0"
#define GPIO_KEYS_0_GPIO_FLAGS          256
#define GPIO_KEYS_0_GPIO_PIN            11
#define GPIO_KEYS_0_LABEL               "Push button switch 0"
#define SW0_GPIO_CONTROLLER             GPIO_KEYS_0_GPIO_CONTROLLER
#define SW0_GPIO_FLAGS                  GPIO_KEYS_0_GPIO_FLAGS
#define SW0_GPIO_PIN                    GPIO_KEYS_0_GPIO_PIN
#define SW0_LABEL                       GPIO_KEYS_0_LABEL

Reply:
Not too bad!

Reply:
@ioannisg They are defined through the aliases in the dts file.

The dts nodes _gpio_keys_ and _gpio_leds_ each generate a define prefixed with the node name, followed by the number after the @ and postfixed with _CONTROLLER, _FLAGS _PIN and _LABEL.

For example: buttonA : @ 0  {...} would generate:
- GPIO_KEYS_0_GPIO_CONTROLLER
- GPIO_KEYS_0_GPIO_FLAGS
- GPIO_KEYS_0_GPIO_LABEL
- GPIO_KEYS_0_GPIO_PIN

An alias replaces the node name and entry number with the alias name.
For example: buttonA = &button1 would generate:
- BUTTON1_GPIO_CONTROLLER
- BUTTON1_GPIO_PIN
and so forth.

That being said, some of the aliases found in board.h can not be generated by dts: for example SW0_GPIO_NAME or LED0_GPIO_PORT; that's because both are generated as _CONTROLLER.
Thus small changes to some examples were necessary.

Reply:
>BUTTON1_GPIO_CONTROLLER
BUTTON1_GPIO_PIN

You mean, SW1_GPIO_PIN, right?

Reply:
Well, that was an example (and there's a mistake in it).
In practice this alias: `sw0 = &button0;` generates a bunch of defines like: `SW0_GPIO_CONTROLLER`, `SW0_GPIO_PIN` etc.